packages/keys/src/key-pair.ts

```diff
@@ -29,21 +29,8 @@ export async function createKeyPairFromBytes(bytes: Uint8Array, extractable?: bo
     const signedData = await signBytes(privateKey, TEST_DATA);
     const isValid = await verifySignature(publicKey, signedData, TEST_DATA);
     if (!isValid) {
-        throw new SolanaError(SOLANA_ERROR__KEYS__VERIFY_SIGNATURE_FAILURE);
+        throw new SolanaError(SOLANA_ERROR__KEYS__PUBLIC_KEY_MUST_MATCH_PRIVATE_KEY, { address: bytes.slice(32) });
```

Comment:
Thanks for the update! Just one more comment. It's probably more useful for the end-user if that address is base-58 encoded.

You can use string codecs to transform it like so:

```ts
import { getStringDecoder, getBase58Decoder } from "@solana/codecs-strings";

const addressDecoder = getStringDecoder({ encoding: getBase58Decoder(), size: 32 });
const address = addressDecoder.decode(bytes.slice(32));
```

Note that you could also use `getAddressDecoder` from `@solana/addresses` but that would add a new dependency to `@solana/keys` when it's not really needed.

Reply:
I actually think we should probably not log the bytes here. Imagine that someone got the order wrong, and now their private bytes are logged to the console. Include no context.

Reply:
That's a very good point.